causalml/inference/meta/rlearner.py

```diff
@@ -271,6 +304,63 @@ def bootstrap(self, X, p, treatment, y, size=10000):
         te_b = self.predict(X=X)
         return te_b
 
+    def run_propensity_score(self, X, treatment, X_pred=None, treatment_pred=None, cv=None,
+                             calibrate_propensity=True, return_model=False):
+        """Generate propensity score if user didn't provide
+
+        Args:
+            X (np.matrix): features for training
+            treatment (np.array or pd.Series): a treatment vector for training
+            X_pred (np.matrix, optional): features for prediction
+            treatment_pred (np.array or pd.Series, optional): a treatment vector for prediciton
+            cv (sklearn.model_selection._BaseKFold, optional): sklearn CV object
+            calibrate_propensity (bool, optional): whether calibrate the propensity score
+            return_model (bool, optional): whether return the propensity model
+        Returns:
+            (numpy.ndarray or numpy.ndarray and dict): propensity score.
+                If return_model, returns propensity score and a dictionary of propensity model
+        """
+        if X_pred is None:
+            p = np.zeros_like(treatment, dtype=float)
+        else:
+            p = np.zeros_like(X_pred, dtype=float)
+
+        p_model = ElasticNetPropensityModel()
+        model_dict = dict()
+
+        if cv:
+            n_fold = 0
+            for i_fold, (i_trn, i_val) in enumerate(cv.split(X, treatment), 1):
+                logger.info('Training a propensity model for CV #{}'.format(i_fold))
+                p_model.fit(X[i_trn], treatment[i_trn])
+                model_dict[i_fold] = p_model
+                if X_pred is None:
+                    p[i_val] = p_model.predict(X[i_val])
+                else:
+                    p_fold = np.zeros_like(treatment_pred, dtype=float)
+                    p_fold += p_model.predict(X_pred)
+                    n_fold += 1
+            p = p_fold/n_fold
```

Comment:
To get the number of folds, we can use `cv. get_n_splits(X)` without using `n_fold`.

Reply:
Thanks for the suggestion here! Changed accordingly.